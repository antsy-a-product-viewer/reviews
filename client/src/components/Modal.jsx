import React from 'react';
import moment from 'moment';
import Review from './Review.jsx';
import ReviewContainer from './ReviewContainer.jsx';
import modalStyles from './css/modalStyles.css.js';

const Modal = (props) => {
  if (!props.showModal) {
    return null;
  } else {
    return (
      <div style={modalStyles.modal} onClick={props.onClose}>
        <section style={modalStyles.modalMain}>
          <img style={modalStyles.image} src="https://i.etsystatic.com/iap/c08960/1735567682/iap_640x640.1735567682_9rntz5h8.jpg?version=0"></img>
          <div style={modalStyles.userInfo}>
            <img style ={modalStyles.profileImg} src={reviews[0].user_img}></img>
            <div style={modalStyles.username}>{reviews[0].user_name}</div>
            <div style={modalStyles.date}>{moment(reviews[0].created_at).format('MMM DD, YYYY')}</div>
            <div>
              {/* <Review review={reviews[0]}/> */}
            </div>
          </div>
          <button style={modalStyles.button} onClick={props.onClose}>
            <img style={modalStyles.buttonImg}src="https://cdn0.iconfinder.com/data/icons/elasto-online-store/26/00-ELASTOFONT-STORE-READY_close-512.png"></img>
          </button>
        </section>
      </div>
    );
  }
};

var reviews = [
  {
    "user_img": "https://s3-us-west-1.amazonaws.com/antsyuserimages/00010.jpg",
    "user_name": "Jalen Paucek",
    "created_at": "2014-02-15T21:43:09.000Z",
    "stars": 5,
    "review": "You can't connect the pixel without overriding the auxiliary RSS card!",
    "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00072.jpg",
    "item_name": "Ergonomic Concrete Tuna"
  },
  {
    "user_img": "https://s3-us-west-1.amazonaws.com/antsyuserimages/00027.jpg",
    "user_name": "Zella Blick",
    "created_at": "2014-03-14T22:37:45.000Z",
    "stars": 4,
    "review": "Try to hack the SQL transmitter, maybe it will navigate the multi-byte bus!",
    "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00030.jpg",
    "item_name": "Handmade Concrete Salad"
  },
  {
    "user_img": "https://s3-us-west-1.amazonaws.com/antsyuserimages/00024.jpg",
    "user_name": "Deangelo Schuppe",
    "created_at": "2014-03-27T15:37:49.000Z",
    "stars": 0,
    "review": "I'll generate the cross-platform COM firewall, that should driver the XSS pixel!",
    "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00091.jpg",
    "item_name": "Rustic Metal Chair"
  },
  {
    "user_img": "https://s3-us-west-1.amazonaws.com/antsyuserimages/00014.jpg",
    "user_name": "Stacey Spencer",
    "created_at": "2014-05-01T06:50:00.000Z",
    "stars": 3,
    "review": "We need to reboot the primary XML circuit!",
    "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00004.jpg",
    "item_name": "Generic Steel Soap"
  }
]

export default Modal;