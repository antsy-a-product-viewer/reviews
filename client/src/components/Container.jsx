import React from 'react';
import axios from 'axios';
import _ from 'underscore';
import Star from './Star.jsx';
import ReviewPhotosContainer from './ReviewPhotosContainer.jsx';
import Modal from './Modal.jsx';
import containerStyles from './css/containerStyles.css.js';
import ReviewContainer from './ReviewContainer.jsx';
import Button from './Button.jsx';

class Container extends React.Component {
  constructor(props) {
    super(props);
    this.openModal = this.openModal.bind(this);
    this.closeModal = this.closeModal.bind(this);
    this.getData = this.getData.bind(this);
    this.getReviews = this.getReviews.bind(this);
    this.getReviewImages = this.getReviewImages.bind(this);
    this.getAverageStars = this.getAverageStars.bind(this);
    this.showMore = this.showMore.bind(this);
    this.state = {
      storeId: 0,
      averageStars: 0,
      reviews: [],
      reviewImages: [],
      showModal: false,
      show: 4,
      modalReview: null
    };
  }

  componentDidMount() {
    this.getData();
  }

  openModal(reviewImgIndex) {
    this.setState({
      modalReview: reviewImgIndex,
      showModal: !this.state.showModal
    });
  }

  closeModal(e) {
    console.log(`e.target.id: ${e.target.id}`);
    if (e.target.id === 'modal-overlay' || e.target.id === 'close-modal') {
      this.setState({
        showModal: !this.state.showModal,
        modalReview: null
      });
    }
  }

  getData() {
    var itemId = Math.floor(Math.random() * 100);
    axios.get(`/items/${itemId}`)
      .then((res) => {
        var storeId = res.data[0].store_id;
        this.setState({
          storeId: storeId
        });
        this.getReviews(storeId);
        this.getReviewImages(storeId);
      })
      .catch((err) => {
        console.log(err);
      });
  }

  getReviews(storeId) {
    axios.get(`/stores/${storeId}/reviews`)
      .then((res) => {
        this.getAverageStars(res.data);
        this.setState({
          reviews: res.data
        });
      })
      .catch((err) => {
        console.log(err);
      }); 
  }

  getReviewImages(storeId) {
    axios.get(`/stores/${storeId}/review_images`)
      .then((res) => {
        this.setState({
          reviewImages: res.data
        });
      })
      .catch((err) => {
        console.log(err);
      });
  }

  getAverageStars(reviews) {
    var total = 0;
    for (var i = 0; i < reviews.length; i++) {
      total += reviews[i].stars;
    }
    var average = Math.floor(total / reviews.length);
    this.setState({
      averageStars: average
    });
  }

  showMore() {
    this.setState({
      show: 20
    });
  }
  
  render() {
    return (
      <div style={containerStyles.container}>
        <div style={{display: 'flex'}}>
          <div style={containerStyles.header}>
            Reviews
          </div>
          <Star stars={this.state.averageStars}/>
          <div style={containerStyles.centeredDiv}>
            ({this.state.reviews.length})
          </div>
        </div>
        <ReviewContainer reviews={this.state.reviews} limit={this.state.show} showPrice="false"/>
        <Button currentNumber={this.state.show} showMore={this.showMore} totalReviews={this.state.reviews.length}/>
        <ReviewPhotosContainer reviewImages={this.state.reviewImages} openModal={this.openModal}/>
        <Modal showModal={this.state.showModal} onClose={this.closeModal} review={this.state.reviewImages[this.state.modalReview]}/>
      </div>
    );
  }
}


export default Container;