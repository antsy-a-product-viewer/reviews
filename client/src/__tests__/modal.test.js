import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';
import Modal from '../components/Modal.jsx';
import Container from '../components/Container.jsx'

configure({ adapter: new Adapter() });

it('should render correctly when props.showModal is true', () => {
  const props = {
    review: reviewImages[0],
    showModal: true
  };
  const modal = mount(<Modal {...props}/>);
  
  expect(modal).toMatchSnapshot();
});

it('should display the modal when showModal state is true', () =>{
  const props = {
    review: reviewImages[0],
    showModal: true
  };
  const modal = mount(<Modal {...props}/>);

  expect(modal.find('.modal').exists()).toEqual(true);
});

it('should not display the modal when showModal state is false', () =>{
  const props = {
    review: reviewImages[0],
    showModal: false
  };
  const modal = mount(<Modal {...props}/>);

  expect(modal.find('.modal').exists()).toEqual(false);
});

const reviewImages = [
  {
      "review_id": 896,
      "user_img": "https://s3-us-west-1.amazonaws.com/antsyuserimages/00002.jpg",
      "user_name": "Santina Kulas",
      "created_at": "2014-04-18T13:22:26.000Z",
      "stars": 4,
      "review": "The USB card is down, program the haptic sensor so we can generate the IB microchip!",
      "review_img": null,
      "item_id": 58,
      "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00058.jpg",
      "item_name": "Gorgeous Fresh Mouse",
      "price": 922
  },
  {
      "review_id": 764,
      "user_img": "https://s3-us-west-1.amazonaws.com/antsyuserimages/00016.jpg",
      "user_name": "Santa Predovic",
      "created_at": "2014-05-03T17:17:53.000Z",
      "stars": 3,
      "review": "You can't bypass the firewall without navigating the multi-byte AI port!",
      "review_img": null,
      "item_id": 58,
      "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00058.jpg",
      "item_name": "Gorgeous Fresh Mouse",
      "price": 922
  },
  {
      "review_id": 347,
      "user_img": "https://s3-us-west-1.amazonaws.com/antsyuserimages/00013.jpg",
      "user_name": "Adela Prosacco",
      "created_at": "2014-05-04T23:39:10.000Z",
      "stars": 1,
      "review": "We need to program the cross-platform COM feed!",
      "review_img": null,
      "item_id": 92,
      "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00092.jpg",
      "item_name": "Rustic Cotton Soap",
      "price": 122
  },
  {
      "review_id": 587,
      "user_img": "https://s3-us-west-1.amazonaws.com/antsyuserimages/00004.jpg",
      "user_name": "Dr. Ford Lehner",
      "created_at": "2014-08-08T15:29:40.000Z",
      "stars": 1,
      "review": "We need to hack the primary TCP port!",
      "review_img": null,
      "item_id": 92,
      "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00092.jpg",
      "item_name": "Rustic Cotton Soap",
      "price": 122
  },
  {
      "review_id": 412,
      "user_img": "https://s3-us-west-1.amazonaws.com/antsyuserimages/00007.jpg",
      "user_name": "Lesley Kirlin",
      "created_at": "2014-08-30T22:47:57.000Z",
      "stars": 2,
      "review": "You can't bypass the monitor without calculating the bluetooth RAM pixel!",
      "review_img": null,
      "item_id": 61,
      "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00061.jpg",
      "item_name": "Licensed Concrete Mouse",
      "price": 238
  }
];
