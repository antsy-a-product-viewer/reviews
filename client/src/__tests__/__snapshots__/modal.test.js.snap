// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly when props.showModal is true 1`] = `
<Modal
  review={
    Object {
      "created_at": "2014-04-18T13:22:26.000Z",
      "item_id": 58,
      "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00058.jpg",
      "item_name": "Gorgeous Fresh Mouse",
      "price": 922,
      "review": "The USB card is down, program the haptic sensor so we can generate the IB microchip!",
      "review_id": 896,
      "review_img": null,
      "stars": 4,
      "user_img": "https://s3-us-west-1.amazonaws.com/antsyuserimages/00002.jpg",
      "user_name": "Santina Kulas",
    }
  }
  showModal={true}
>
  <div
    className="modal"
    id="modal-overlay"
    style={
      Object {
        "alignItems": "center",
        "animation": "show .5s ease",
        "backgroundColor": "rgba(0, 0, 0, 0.7)",
        "bottom": 0,
        "display": "flex",
        "justifyContent": "center",
        "left": 0,
        "opacity": 1,
        "overflowX": "hidden",
        "overflowY": "auto",
        "padding": "1rem",
        "position": "fixed",
        "right": 0,
        "top": 0,
        "zIndex": "9999",
      }
    }
  >
    <section
      style={
        Object {
          "background": "white",
          "display": "flex",
          "height": 565,
          "left": "50%",
          "position": "fixed",
          "top": "45%",
          "transform": "translate(-50%,-50%)",
          "width": "65%",
        }
      }
    >
      <div>
        <img
          src={null}
          style={
            Object {
              "height": 565,
              "width": 565,
            }
          }
        />
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-between",
          }
        }
      >
        <div
          className="modalReview"
          style={
            Object {
              "marginTop": 20,
            }
          }
        >
          <div
            style={
              Object {
                "display": "flex",
                "margin": 20,
              }
            }
          >
            <img
              src="https://s3-us-west-1.amazonaws.com/antsyuserimages/00002.jpg"
              style={
                Object {
                  "borderRadius": 50,
                  "height": 50,
                  "width": 50,
                }
              }
            />
            <div
              style={
                Object {
                  "alignSelf": "center",
                  "margin": 5,
                  "marginLeft": 15,
                  "textDecoration": "underline",
                }
              }
            >
              Santina Kulas
            </div>
            <div
              style={
                Object {
                  "alignSelf": "center",
                  "margin": 5,
                }
              }
            >
              Apr 18, 2014
            </div>
          </div>
          <div
            style={
              Object {
                "margin": 20,
              }
            }
          >
            <Star
              stars={4}
            >
              <div
                style={
                  Object {
                    "alignSelf": "center",
                  }
                }
              >
                <img
                  key="0"
                  src="https://s3-us-west-1.amazonaws.com/anstyicons/icon-star-512.png"
                  style={
                    Object {
                      "height": "20px",
                      "width": "20px",
                    }
                  }
                />
                <img
                  key="1"
                  src="https://s3-us-west-1.amazonaws.com/anstyicons/icon-star-512.png"
                  style={
                    Object {
                      "height": "20px",
                      "width": "20px",
                    }
                  }
                />
                <img
                  key="2"
                  src="https://s3-us-west-1.amazonaws.com/anstyicons/icon-star-512.png"
                  style={
                    Object {
                      "height": "20px",
                      "width": "20px",
                    }
                  }
                />
                <img
                  key="3"
                  src="https://s3-us-west-1.amazonaws.com/anstyicons/icon-star-512.png"
                  style={
                    Object {
                      "height": "20px",
                      "width": "20px",
                    }
                  }
                />
              </div>
            </Star>
            <div>
              The USB card is down, program the haptic sensor so we can generate the IB microchip!
            </div>
          </div>
        </div>
        <div
          className="modalItem"
        >
          <div
            style={
              Object {
                "margin": 20,
              }
            }
          >
            <Item
              review={
                Object {
                  "created_at": "2014-04-18T13:22:26.000Z",
                  "item_id": 58,
                  "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00058.jpg",
                  "item_name": "Gorgeous Fresh Mouse",
                  "price": 922,
                  "review": "The USB card is down, program the haptic sensor so we can generate the IB microchip!",
                  "review_id": 896,
                  "review_img": null,
                  "stars": 4,
                  "user_img": "https://s3-us-west-1.amazonaws.com/antsyuserimages/00002.jpg",
                  "user_name": "Santina Kulas",
                }
              }
              showPrice="true"
            >
              <div
                style={
                  Object {
                    "display": "flex",
                    "marginTop": 10,
                  }
                }
              >
                <img
                  id={58}
                  onClick={[Function]}
                  src="https://s3-us-west-1.amazonaws.com/antsyitemsimages/00058.jpg"
                  style={
                    Object {
                      "alignSelf": "center",
                      "cursor": "pointer",
                      "height": 75,
                      "width": 75,
                    }
                  }
                />
                <div
                  style={
                    Object {
                      "cursor": "pointer",
                      "margin": "25px",
                    }
                  }
                >
                  <div
                    id={58}
                    onClick={[Function]}
                  >
                    Gorgeous Fresh Mouse
                  </div>
                  <div>
                    $
                    922
                    .00
                  </div>
                </div>
              </div>
            </Item>
          </div>
        </div>
      </div>
      <button
        className="closeModal"
        style={
          Object {
            "background": "none",
            "border": "none",
            "cursor": "pointer",
            "display": "block",
            "float": "right",
            "padding": "none",
            "position": "absolute",
            "right": 5,
            "top": 5,
            "zIndex": 3,
          }
        }
      >
        <img
          id="close-modal"
          src="https://cdn0.iconfinder.com/data/icons/elasto-online-store/26/00-ELASTOFONT-STORE-READY_close-512.png"
          style={
            Object {
              "background": "white",
              "height": 20,
              "margin": 10,
              "width": 20,
            }
          }
        />
      </button>
    </section>
  </div>
</Modal>
`;
